packages/fx-core/tests/component/driver/aad/aadAppClient.test.ts

```diff
@@ -312,6 +314,56 @@ describe("AadAppClient", async () => {
         });
     });
 
+    it("should throw error when CredentialInvalidLifetimeAsPerAppPolicy error happens", async () => {
+      const expectedError = {
+        error: {
+          code: "CredentialInvalidLifetimeAsPerAppPolicy",
+        },
+      };
+
+      const mock = new MockAdapter(axiosInstance);
+      mock
+        .onPost(`https://graph.microsoft.com/v1.0/applications/${expectedObjectId}/addPassword`)
+        .reply(400, expectedError);
+
+      await expect(
+        aadAppClient.generateClientSecret(expectedObjectId)
+      ).to.eventually.be.rejected.then((err) => {
+        expect(err instanceof CredentialInvalidLifetimeError).to.be.true;
+        expect(err.source).equals("AadAppClient");
+        expect(err.name).equals("CredentialInvalidLifetime");
+        expect(err.message).equals(
+          "The client secret lifetime exceeds the max value allowed in your tenant. Set a shorter lifetime using the clientSecretExpireDays parameter."
+        );
+        expect(err.helpLink).equals("https://aka.ms/teamsfx-actions/aadapp-create");
+      });
+    });
+
+    it("should throw error when CredentialTypeNotAllowedAsPerAppPolicy error happens", async () => {
+      const expectedError = {
+        error: {
+          code: "CredentialTypeNotAllowedAsPerAppPolicy",
+        },
+      };
+
+      const mock = new MockAdapter(axiosInstance);
+      mock
+        .onPost(`https://graph.microsoft.com/v1.0/applications/${expectedObjectId}/addPassword`)
+        .reply(400, expectedError);
+
+      await expect(
+        aadAppClient.generateClientSecret(expectedObjectId)
+      ).to.eventually.be.rejected.then((err) => {
+        expect(err instanceof ClientSecretNotAllowedError).to.be.true;
+        expect(err.source).equals("AadAppClient");
+        expect(err.name).equals("ClientSecretNotAllowed");
+        expect(err.message).equals(
+          "Your tenant does not allow creating client secret for Microsoft Entra app. Please create and configure the Microsoft Entra app manually."
```

Comment:
Your tenant doesn't allow creating client secret for Microsoft Entra app. Create and configure the app manually.